mutiny-core/src/lib.rs

```diff
@@ -2189,6 +2192,50 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and withdrawls/redeem/melts the token from it.
+    pub async fn melt_cashu_token(
+        &self,
+        token_v3: TokenV3,
+    ) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let mint_client = CrossPlatformHttpClient::new();
+        let mut invoices: Vec<MutinyInvoice> = Vec::with_capacity(token_v3.tokens.len());
+
+        for token in token_v3.tokens {
+            let mint_url = match token.mint {
+                Some(url) => url,
+                None => return Err(MutinyError::EmptyMintURLError),
+            };
+
+            let invoice_amount = token.proofs.total_amount() as f64 * (1.0 - 3.0 / 100.0);
+
+            let mutiny_invoice = self
+                .create_invoice(invoice_amount as u64, vec!["Cashu Token Melt".to_string()])
+                .await?;
+
+            let mutiny_invoice_str = mutiny_invoice
+                .clone()
+                .bolt11
+                .expect("The invoice should have BOLT11");
+
+            let melt_quote_res = mint_client
+                .post_melt_quote_bolt11(
+                    &mint_url,
+                    mutiny_invoice_str.to_string(),
+                    CurrencyUnit::Sat,
+                )
+                .await?;
+
+            let post_melt_bolt11_response = mint_client
+                .post_melt_bolt11(&mint_url, token.proofs, melt_quote_res.quote, vec![])
+                .await?;
+
+            if post_melt_bolt11_response.paid {
+                invoices.push(mutiny_invoice);
```

Comment:
should refetch the `mutiny_invoice` here so it has all the filled in metadata after it is paid.

should be `self.get_invoice`